libsplinter/src/admin/store/diesel/operations/fetch_circuit.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides the "fetch circuit" operation for the `DieselAdminServiceStore`.
+
+use diesel::{prelude::*, result::Error::NotFound};
+use std::convert::TryFrom;
+
+use super::{list_services::AdminServiceStoreListServicesOperation, AdminServiceStoreOperations};
+use crate::admin::store::{
+    diesel::{
+        models::{CircuitMemberModel, CircuitModel},
+        schema::{circuit, circuit_member},
+    },
+    error::AdminServiceStoreError,
+    AuthorizationType, Circuit, CircuitBuilder, DurabilityType, PersistenceType, RouteType,
+    Service,
+};
+
+pub(in crate::admin::store::diesel) trait AdminServiceStoreFetchCircuitOperation {
+    fn fetch_circuit(&self, circuit_id: &str) -> Result<Option<Circuit>, AdminServiceStoreError>;
```

Comment:
Bit of a change-up, but I think we decided to go with `get_*` instead of `fetch_*` for stores.

Reply:
Ok, since that is an admin store API change, perhaps that should go in another PR? I don't think that change would fit with this db implementation (and this pr is big enough as-is)